src/Particular.LicensingComponent/ThroughputCollector.cs

```diff
@@ -89,19 +89,33 @@ public async Task<List<EndpointThroughputSummary>> GetThroughputSummary(Cancella
         return endpointSummaries;
     }
 
-    public async Task<ReportGenerationState> GetReportGenerationState(CancellationToken cancellationToken)
+    public async Task<ReportGenerationState> GetReportGenerationState(CancellationToken cancellationToken) =>
+        throughputQuery == null ?
+            await GetReportGenerationStateForNonBroker(cancellationToken) :
+            await GetReportGenerationStateForBroker(cancellationToken);
+
+    async Task<ReportGenerationState> GetReportGenerationStateForBroker(CancellationToken cancellationToken)
     {
-        var reportGenerationState = new ReportGenerationState(transport)
+        var reportCanBeGenerated = await dataStore.IsThereThroughputForLastXDaysForSource(30, ThroughputSource.Broker, false, cancellationToken);
+        var reason = reportCanBeGenerated ? string.Empty : $"Need at least one day of usage data in the last 30 days from ${transport} broker.";
```

Comment:
`transport` prop here, will display an unformatted string, eg `RabbitMQ.ClassicConventionalRouting`
In SP we format these strings, see https://github.com/Particular/ServicePulse/blob/ca4c4c8700a53051052151a13a0e86ef21189d83/src/Frontend/src/stores/ThroughputStore.ts#L54

Reply:
has anyone tested the length of this string in SP? What does it look like with a long transport name?

Reply:
Thank you, nice spot.  I actually meant to get the transport from the broker query which I believe is short form.